Association_Mining/blog.md

```diff
@@ -0,0 +1,219 @@
+Given a support threshold *'s'*, sets of items that appear in at least *s* baskets are called as **frequent item sets**. In this example, let us assume the support threshold is 3.
+We will count the frequency of the singleton set, i.e. a single item and decide which of them can be included in the frequent item sets.
+
+|Item| Count/Support|
+|---|---|
+|M|5|
+|C|5|
+|P|2|
+|B|6|
+|J|4|
+
+Here, only 'Pepsi' has a count of 2, which is below the support threshold. Hence, we can say that the frequent item sets of size 1 are {M},{C},{B} and {J}.
+Similarly, the frequent item sets of size 2 are {M,B}, {C,B} and {C,J} and frequent item set of size 3 is {M,C,B}.
+We now calculate the confidence of the association rules which we can form using the frequent item sets. Assume that the threshold for confidence is 0.75.
+Using the formula given above, we calculate the confidence of the rules:
+B→M has a confidence of 4/6. This is below the threshold and we do not consider this rule. However, the rule B→C has a confidence of 5/6 and it is above the confidence threshold level, so we consider this as one of the association rule. This is how the association rules are generated.
+
+Now, we turn to the practical application of this algorithm on a given dataset.
+(Audiences are encouraged to use various tools to have their implementations, such as Goole Colab, Jupyter notebook)
+
+The data which we will work on, is a real online retail transaction data set of two years which is available on this [link](https://archive.ics.uci.edu/ml/datasets/Online+Retail+II).
+The data has 1067371 instances and 8 features. The downloaded dataset is in an excel file format.
+Let us first start with importing the libraries which are required to run the program.
+
+##### Import Libraries
+
+```
+import pandas as pd
+import numpy as np
+from mlxtend.frequent_patterns import apriori, association_rules
+```
+
+In the script above we import pandas, numpy, apriori and association_rules libraries.
+
+Some audiences may encounter problems like missing modules. To solve, simply install the libraries using ``` pip install <library name> ``` or follow the instructions given in the documentations of the respective libraries to install the required components.
+
+##### Importing the Dataset
+
+Now let's import the dataset and see what we are working with. Download the dataset and place it in the *data* folder  and execute the following script:
+
+```
+data = pd.read_excel("data/online_retail_II.xlsx")
+data_temp = pd.read_excel("data/online_retail_II.xlsx", 'Year 2010-2011')
+```
+
+If you open the excel file, you will notice that the data is separated into years by different worksheets. Hence, we import both the worksheets and combine them into a single data frame using the following code:
+
+```
+data = pd.concat([data,data_temp])
+```
+
+To see the first 5 rows of the data:
+
+```
+data.head()
+```
+
+The data frame will look like this:
+
+![data_head output](/images/data_head.png)

```

Comment:
This image doesn't show up when viewing the file

Reply:
changed the file path for the image.